### PR TITLE
feat: controller init

### PR DESCRIPTION
**Pull request by SnowFireWolf**
# Description

* Add controller to routes
* modify user update

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: src/router/api/authentication/controller.ts [hunk 1]
```diff
@@ -0,0 +1,297 @@
+import { isUserLocale } from '@/model/auth/user_locale';
+import { ConnectType } from '@/model/auth/connect_account';
+import { OauthData } from '#/api/authentication/oauth_data';
+import { HttpStatusCode, ResponseStatusCode, sendResponse } from '#/util';
+import { Request, Response } from 'express';
+import { connectOAuthAccount } from '#/api/authentication/util';
+import { getIp } from '#/util/util';
+import { User } from '@/model/auth/user';
+
+export const googleAuth = async (req: Request, res: Response) => {
```
Comment (SiongSng) [inline src/router/api/authentication/controller.ts]: 應該是 `googleOAuth` 才對，其他 function 亦同 (google/facebook)

**Review by SiongSng — CHANGES_REQUESTED** [review]
Looks good.
just a little problem.

File: src/router/api/user/controller.ts [hunk 2]
```diff
@@ -0,0 +1,461 @@
...
+        {
+          code: ResponseStatusCode.USER_NOT_FOUND,
+        },
+        HttpStatusCode.NOT_FOUND
+      );
+    }
+  }
+};
+
+export const updateInfo = async (req: Request, res: Response) => {
```
Comment (SiongSng) [inline src/router/api/user/controller.ts:83]:
```suggestion
export const editInfo = async (req: Request, res: Response) => {
```
應該是 `editInfo`

**Review by SiongSng — CHANGES_REQUESTED** [review]

File: src/router/api/user/data.d.ts [hunk 3]
```diff
@@ -1,29 +1,29 @@
...
- */
-export interface LoginUserData {
-  email: string;
-  password: string;
-}
+import { UserLocale } from '@/model/auth/user_locale';
+import { UserMode } from '@/model/auth/user_mode';
+
+/**
+ * The data for editing user info.
```
Comment (SiongSng) [inline src/router/api/user/data.d.ts]:
```suggestion
 * The data for updating user info.
```

File: src/router/api/user/controller.ts [hunk 4]
```diff
@@ -0,0 +1,461 @@
...
+  /*
+    #swagger.description = 'Edit the info of the current user (authorization required)'
```
Comment (SiongSng) [inline src/router/api/user/controller.ts]:
```suggestion
    #swagger.description = 'Update the info of the current user (authorization required)'
```

**Review by SiongSng — CHANGES_REQUESTED** [review]

**Comment by lumynou5** [comment]
Sorry, but did you read the guideline?

**Review by SiongSng — APPROVED** [review]
LGTM

**Pull request merged**